sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRepositoryAsyncClient.java

```diff
@@ -0,0 +1,650 @@
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return tag attributes by tag.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<TagProperties>> getTagPropertiesWithResponse(String tag) {
+        return withContext(context -> getTagPropertiesWithResponse(tag, context));
+    }
+
+    Mono<Response<TagProperties>> getTagPropertiesWithResponse(String tag, Context context) {
+        try {
+            if (tag == null) {
+                return monoError(logger, new NullPointerException("'tag' cannot be null."));
+            }
+            if (context == null) {
+                return monoError(logger, new NullPointerException("'context' cannot be null."));
+            }
+
+            return this.serviceClient.getTagPropertiesWithResponseAsync(repositoryName, tag, context)
+                .map(res -> Utils.mapResponse(res, Utils::mapTagProperties))
+                .onErrorMap(Utils::mapException);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Get tag attributes.
+     *
+     * @param tag Tag name.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return tag attributes by tag.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<TagProperties> getTagProperties(String tag) {
+        return this.getTagPropertiesWithResponse(tag).map(Response::getValue);
+    }
+
+    /**
+     * List tags of a repository.
+     *
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return list of tag details.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<TagProperties> listTags() {
+        return listTags(null);
+    }
+
+    /**
+     * List tags of a repository.
+     *
+     * @param options tagOptions to be used for the given operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws ResourceNotFoundException thrown if the given digest was not found.
+     * @return list of tag details.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<TagProperties> listTags(ListTagsOptions options) {
+        return new PagedFlux<>(
+            (pageSize) -> withContext(context -> listTagsSinglePageAsync(pageSize, options, context)),
```

Comment:
How does the user set the pageSize?

Reply:
by calling the listTags.byPage(pageSize) --> it is one of the overloads of byPage(int pageSize).

---
In reply to: [609640147](https://github.com/Azure/azure-sdk-for-java/pull/20368#discussion_r609640147) [](ancestors = 609640147)